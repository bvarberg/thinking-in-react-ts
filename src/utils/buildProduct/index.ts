import { Product } from "../../types/Product"

type BuildProduct = () => Product

export const buildProduct: BuildProduct = () => {
  return {
    category: "Sporting Goods",
    name: "Football",
    price: "$49.99",
    stocked: true,
  }
}
