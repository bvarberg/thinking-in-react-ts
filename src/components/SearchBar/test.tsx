import { SearchBar } from "."
import React from "react"
import { shallowRender } from "../../utils/shallowRender"

describe("SearchBar", () => {
  it("renders", () => {
    const received = shallowRender(<SearchBar />)

    expect(received).toMatchSnapshot()
  })

  describe("with in stock only", () => {
    it("renders", () => {
      const inStockOnly = true

      const received = shallowRender(<SearchBar inStockOnly={inStockOnly} />)

      expect(received).toMatchSnapshot()
    })
  })

  describe("with filter text", () => {
    it("renders", () => {
      const filterText = "Football"

      const received = shallowRender(<SearchBar filterText={filterText} />)

      expect(received).toMatchSnapshot()
    })
  })
})
