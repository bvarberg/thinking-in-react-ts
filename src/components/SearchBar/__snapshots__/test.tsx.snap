// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar renders 1`] = `
<form
  className="SearchBar"
>
  <input
    placeholder="Search..."
    type="text"
  />
  <label
    htmlFor="onlyShowInStock"
    style={
      Object {
        "display": "block",
      }
    }
  >
    <input
      name="onlyShowInStock"
      type="checkbox"
    />
    Only show products in stock
  </label>
</form>
`;
